cluster/module/server.c

```diff
@@ -724,9 +748,158 @@ _alert_service_with_zeroed_score(struct conscience_srv_s *srv)
 	}
 }
 
+/* -------------------------------------------------------------------------- */
+
+static struct service_info_dated_s *
+service_info_dated_new2(struct conscience_srv_s *srv)
+{
+	struct service_info_dated_s *sid = g_malloc0(sizeof(
+			struct service_info_dated_s));
+	sid->si = g_malloc0(sizeof(struct service_info_s));
+	conscience_srv_fill_srvinfo(sid->si, srv);
+	sid->lock_mtime = srv->lock_mtime;
+	sid->tags_mtime = srv->tags_mtime;
+	return sid;
+}
+
+static struct conscience_srv_s *
+conscience_srvtype_refresh_from_conscience(
```

Comment:
The "_from_conscience" in the name of these functions is confusing. I guess the difference between these functions and the other ones (without the "_from_conscience" suffix) is that they act on "dated" service descriptions, and will do date comparisons before overwriting known service descriptions.